### PR TITLE
Additional extensions and documentation

### PR DESCRIPTION
**Pull request by iloveitaly**
Quick overview:

* More status updates during installation
* Output ./configure options for easy debugging
* Support sodium and gmp extensions
* Lots of documentation updates

I removed `PHP_WITH_GMP` and autodetected the package instead.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,13 +21,54 @@ for us to support them.
...
 ```bash
-asdf plugin-add php https://github.com/asdf-community/asdf-php.git
+asdf plugin-add php
 ```
 
+Since this plugin compiles PHP from source, you'll want to ensure all other required packages are installed.
+After you've installed additional system packages, install a specific PHP version:
+
+```bash
+asdf install php 7.4.16
```
Comment (smorimoto) [inline README.md:31]:
```
asdf install php latest
```

File: README.md [hunk 2]
```diff
@@ -21,13 +21,54 @@ for us to support them.
 ## Installation
 
 ```bash
-asdf plugin-add php https://github.com/asdf-community/asdf-php.git
```
Comment (smorimoto) [inline README.md:24]: Do not change this.

File: README.md [hunk 3]
```diff
@@ -21,13 +21,54 @@ for us to support them.
...
+After you've installed additional system packages, install a specific PHP version:
+
+```bash
+asdf install php 7.4.16
+```
+
+You can test to ensure the installation was successful:
+
+```bash
+asdf reshim
```
Comment (smorimoto) [inline README.md:37]: Reshim should not be needed here.

File: README.md [hunk 4]
```diff
@@ -21,13 +21,54 @@ for us to support them.
...
+
+```bash
+pecl install redis
+pecl install imagick
+
+echo "extension=redis.so
+extension=imagick.so" > $(asdf where php)/conf.d/php.ini
+```
+
+#### MacOS
```
Comment (smorimoto) [inline README.md:51]: macOS

File: README.md [hunk 5]
```diff
@@ -21,13 +21,54 @@ for us to support them.
...
+
+To install PHP on MacOS, you'll need a set of packages installed via homebrew:
+
+```bash
+brew install bison bzip2 freetype gettext libiconv icu4c jpeg libedit libpng libxml2 libzip openssl readline webp zlib
+```
```
Comment (smorimoto) [inline README.md:57]: See here: https://github.com/asdf-community/asdf-php#prerequirements

File: bin/install [hunk 6]
```diff
@@ -80,12 +83,20 @@ install_php() {
...
     ./configure $configure_options || exit 1
+    echo "Running make \"$make_flags\""
     make "$make_flags" || exit 1
+    echo "Running make install..."
     # make "$make_flags" test || exit 1
     make "$make_flags" install || exit 1
   )
+
+  # it's not obvious where php.ini should be placed, let us make it easy for the user
+  mkdir $install_path/conf.d/
```
Comment (smorimoto) [inline bin/install:98]: `mkdir -p` is preferred

**Comment by smorimoto** [comment]
We manage many repositories under this organization and are conscious of reducing duplication of description between repositories whenever possible. For this reason, it's preferred that the usage of asdf itself remain unchanged.

**Comment by iloveitaly** [comment]
Wow, thanks for the quick review!

> For this reason, it's preferred that the usage of asdf itself remain unchanged

What do you mean here? What part of the usage is changed here, the readme?

File: README.md (hunk 5 above)
Comment (iloveitaly) [reply README.md:57]: I did see that, but it feels a bit more user friendly to include the exact command needed for the installation here so the user doesn't have to go digging in a GH workflow config. Happy to drop this if it feels too verbose.

**Comment by smorimoto** [comment]
The Installation section usually describes how to install the plugin itself, and we can assume that the installation instructions for specific versions of PHP are explained by the usage section.

File: README.md (hunk 5 above)
Comment (smorimoto) [reply README.md:56]: I agree with that, but if it becomes obsolete and different from CI, more users will be confused. 
In many cases, we can only guarantee that a weekly scheduled build does not break our plugin. Addition, I often miss that...

**Comment by smorimoto** [comment]
You can fix formatting check errors with `make fmt` after installing `shfmt`.

File: README.md (hunk 5 above)
Comment (iloveitaly) [reply README.md]: Makes sense. I removed the explicit command and just linked to the workflow area. The link contains the master sha, but I think that's an ok tradeoff: users can just switch to the master branch to get the latest brew command.

**Comment by iloveitaly** [comment]
Formatting is fixed and I've updated the readme. Let me know if there are any other tweaks!

**Review by smorimoto — APPROVED** [review]

**Comment by smorimoto** [comment]
Thanks @iloveitaly!

**Pull request merged**